File: back/models/Item.js
const mongoose = require('mongoose');

const ItemCategory = Object.freeze({
	HAT:		'hat',
	TSHIRT:		'tshirt',
	BOOTS:		'boots',
	PANTS:		'pants',
	HAND:		'hand',
	ACCESSORY:	'accessory',
	MISC:		'misc'
});

const ItemSchema = new mongoose.Schema({
    id: { type: String, required: true, unique: true, index: true },
    name: { type: String, required: true, index: true },
    category: { type: String, enum: Object.values(ItemCategory), default: ItemCategory.MISC, index: true },

    permanent: { type: Boolean, default: true },
    price: { type: Number, required: true, min: 0 },

    modelUrl: { type: String },
    description: { type: String, default: '' },
	// Additional metadata can be stored here, that's not explicitly defined.
    metadata: { type: mongoose.Schema.Types.Mixed }
}, { timestamps: true });

const Item = mongoose.model('Item', ItemSchema);
module.exports = Item;
module.exports.ItemCategory = ItemCategory;

File: back/models/item.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

type Item struct {
	ID          string         `gorm:"type:varchar(255);primaryKey"`
	Name        string         `gorm:"type:varchar(255);not null;index"`
	Category    string         `gorm:"type:varchar(100);default:'misc';index"`
	IsSellable  bool           `gorm:"not null;default:false"`
	Price       int            `gorm:"not null;default:100"`
	Description string         `gorm:"type:text"`
	Image       string         `gorm:"type:varchar(255)"`
	Rarity      string         `gorm:"type:varchar(50);default:'common'"`
	Attributes  datatypes.JSON `gorm:"type:jsonb"`
	CreatedAt   time.Time
	UpdatedAt   time.Time
}
